### ISSUE TITLE
License exception for static linking

### ISSUE THREAD

**Issue by pombredanne** [thomasmoelhave/tpie#242]
From what I can see the default is to build the library for static linking per https://github.com/thomasmoelhave/tpie/blob/7c11771f6320c50ab59f795e85a1d7c6e69b06e1/CMakeLists.txt#L72

Typically, this may be a problem with the LGPL, as the caller may be subject to the LGPL too. The way this is handled is usually with an exception. For instance this exception from Canonical:
https://raw.githubusercontent.com/go-xmlpath/xmlpath/v2/LICENSE

    Copyright (c) 2013-2014 Canonical Inc.
    
    This software is licensed under the LGPLv3, included below.
    
    As a special exception to the GNU Lesser General Public License version 3
    ("LGPL3"), the copyright holders of this Library give you permission to
    convey to a third party a Combined Work that links statically or dynamically
    to this Library without providing any Minimal Corresponding Source or
    Minimal Application Code as set out in 4d or providing the installation
    information set out in section 4e, provided that you comply with the other
    provisions of LGPL3 and provided that you meet, for the Application the
    terms and conditions of the license(s) which apply to the Application.
    
    Except as stated in this special exception, the provisions of LGPL3 will
    continue to comply in full to this Library. If you modify this Library, you
    may apply this exception to your version of this Library, but you are not
    obliged to do so. If you do not wish to do so, delete this exception
    statement from your version. This exception does not (and cannot) modify any
    license terms which apply to the Application, with which you must still
    comply.

Would you consider granting such exception here too?

**Comment by antialize** [comment]
It is not really feasible for us to change the license. If you want to ship software that uses TPIE and not release that software under GPL or LGPL v3, you should build tpie as a shared library.

**Comment by pombredanne** [comment]
@antialize Thank you for the quick reply! Can you elaborate a bit on the "feasible"?

**Comment by antialize** [comment]
We do not require that contributes to TPIE hand over the copyright of their contributions to some TPIE organization, so to change the license would require us to get permission from each an every committer. We did that when we changed from GPLV2 to LGPLv3, but that was a very complicated process that I and none of the other maintainers want to do again.

This project is OLD, the first commit is from 1994, but the code is older still. Some of the authors have not touched the project in decades, and are quite busy with other things.
